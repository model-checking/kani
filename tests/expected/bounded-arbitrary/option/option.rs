// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT

//! This file tests whether we can generate a bounded option (`T: BoundedArbitrary` in `Option<T>`) that correctly
//! represents None or Some(t) where t is bounded by N.

#[kani::proof]
fn check_option() {
    let my_option: Option<Vec<bool>> = kani::bounded_any::<_, 4>();
    kani::cover!(my_option.is_none());
    if let Some(inner) = my_option {
        kani::cover!(inner.len() == 0);
        kani::cover!(inner.len() == 1);
        kani::cover!(inner.len() == 2);
        kani::cover!(inner.len() == 3);
        kani::cover!(inner.len() == 4);
    }
}
