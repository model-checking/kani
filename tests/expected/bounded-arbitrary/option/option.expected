Checking harness check_option...

** 6 of 6 cover properties satisfied

Complete - 1 successfully verified harnesses, 0 failures, 1 total.